tests/integration/wallet/sync.spec.js

```diff
@@ -222,7 +223,126 @@ describe('Wallet sync tests', () => {
         await mineBlocks(1);
         expect(walletHD.getCurrentAddress()).toBe(newAddress);
     });
+    it('correctly updates historical txs', async () => {
+        // Keep track of previous historical txs
+        let prevIds = walletHD.getHistoricalTxs().map((hTx) => {
+            return hTx.id;
+        });
+        /**
+         * @type {HistoricalTx[]}
+         */
+        let diffCache = [].concat(walletHD.getHistoricalTxs());
+
+        // 1) Receive a transaction to the current address
+        let addr = walletHD.getCurrentAddress();
+        let blockHeight = getNetwork().getBlockCount() + 1;
+        await createAndSendTransaction(walletLegacy, addr, 0.01 * 10 ** 8);
+        // Sanity check: Before mining the block the history is unchanged (receiving from external wallet)
+        getHistDiff(walletHD.getHistoricalTxs(), prevIds, 0);
+        checkHistPersistence(walletHD.getHistoricalTxs(), diffCache);
+        // Mine and assert that a new tx has been added
+        await mineBlocks(1);
+        let diff = getHistDiff(walletHD.getHistoricalTxs(), prevIds, 1);
+        checkHistPersistence(walletHD.getHistoricalTxs(), diffCache);
```

Comment:
For readability sake, can you linebreak each of the `x)` steps and other 'sectioned' code parts? It is a little difficult to read formatted closely like this.